diff --git a/dist/cjs/utils.js b/dist/cjs/utils.js
index b054917807ea6bd25c934a342aa2db8474120899..8ded6c7db8b586bb9b36991735c4a684a30e56d1 100644
--- a/dist/cjs/utils.js
+++ b/dist/cjs/utils.js
@@ -152,7 +152,14 @@ exports.asyncSend = asyncSend;
 const childSend = (proc, msg) => (0, exports.asyncSend)(proc, msg);
 exports.childSend = childSend;
 const isRedisVersionLowerThan = (currentVersion, minimumVersion) => {
+    if (typeof currentVersion === 'object') {
+        currentVersion = currentVersion.version || '';
+    }
+
     const version = semver.valid(semver.coerce(currentVersion));
+    if (!version) {
+        throw new TypeError(`Invalid version. Must be a string. Got type "${typeof currentVersion}".`);
+    }
     return semver.lt(version, minimumVersion);
 };
 exports.isRedisVersionLowerThan = isRedisVersionLowerThan;
